src/duplicacy_utils.go

```diff
@@ -325,24 +340,25 @@ func MatchPath(filePath string, patterns [] string) (included bool) {
              if matchPattern(filePath, pattern[1:]) {
                 return false
             }
-        } else if strings.HasPrefix(pattern, "i:") {
-            matched, err := regexp.MatchString(pattern[2:], filePath)
-            if err != nil {
-                LOG_ERROR("SNAPSHOT_MATCH", "Error during regexp match: %s - %v", pattern, err)
-            }
-            if matched {
-                LOG_TRACE("SNAPSHOT_MATCH", "Regex include comparison for filePath=\"%s\", pattern=\"%s\", matched=%t", filePath, pattern[2:], matched)
-                return true
-            }
-        } else if strings.HasPrefix(pattern, "e:") {
-            allIncludes = false
-            matched, err := regexp.MatchString(pattern[2:], filePath)
-            if err != nil {
-                LOG_ERROR("SNAPSHOT_MATCH", "Error during regexp match: %s - %v", pattern, err)
+        } else if strings.HasPrefix(pattern, "i:") || strings.HasPrefix(pattern, "e:") {
+            if re, found = RegularExpressions[pattern[2:]]; found {
+                LOG_DEBUG("REGEX_FOUND", "Compiled regex for pattern \"%s\" found", pattern)
+                matched = re.MatchString(filePath)
+            } else {
+                LOG_DEBUG("REGEX_NOTFOUND", "Compiled regex for pattern \"%s\" not found!", pattern)
+                matched, _ = regexp.MatchString(pattern[2:], filePath)
             }
+            LOG_DEBUG("REGEX_RESULT", "Regex comparison for filePath=\"%s\", pattern=\"%s\", matched=%t", filePath, pattern, matched)
             if matched {
-                LOG_TRACE("SNAPSHOT_MATCH", "Regex exclude comparison for filePath=\"%s\", pattern=\"%s\", matched=%t", filePath, pattern[2:], matched)
-                return false
+                if strings.HasPrefix(pattern, "i:") {
```

Comment:
Just `return strings.HasPrefix(pattern, "i:")` instead of the complicated if statement.

Reply:
duh, thanks... :)